docs/spec.md

```diff
@@ -2437,8 +2440,8 @@ the XLA compiler. In the future, we are planning to unify this metadata
 %results = "stablehlo.custom_call"(%input0) {
   call_target_name = "foo",
   has_side_effect = false,
-  backend_config = "bar",
-  api_version = 1 : i32,
+  backend_config = {bar = 42 : i32},
+  api_version = 4 : i32,
```

Comment:
This change doesn't seem related to the backend_config change. Consider reverting it.

Reply:
`4` for is new  API_VERSION_TYPED_FFI enum  at https://github.com/openxla/stablehlo/blob/main/stablehlo/dialect/StablehloEnums.td#L88  
For `API_VERSION_TYPED_FFI` custom calls `backend_config` must be a dictionary attribute

Reply:
`API_VERSION_TYPED_FFI` is not defined in `StablehloEnums.td`. Did you mean to link to the MHLO definition?

Consider adding a comment explaining this, magic constants are hard to understand.

Reply:
Thanks for the feedback, added a comment to explain "api_version" constant.